src/main/java/ro/hiringsystem/model/MailInformation.java

```diff
@@ -0,0 +1,34 @@
+package ro.hiringsystem.model;
+
+import lombok.*;
+
+import java.util.UUID;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class MailInformation {
+
+    private UUID id;
+
+    private String firstName;
```

Comment:
Redundant storage of firstName and lastName, this data is already stored inside the User and it is highly unlikely that we will use a mail without the attached user.

Reply:
I agree. The only reason I initially thought this class would be useful is the encapsulation of the checks. But, as we can use annotations for validations, there is no point in keeping this model.